### ISSUE TITLE
Incompatible override for type of matching overrride

### ISSUE THREAD

**Issue by stevenh** [microsoft/pyright#10193]
When creating a class that returns an AsyncIterator which implements the defined method of its abstract parent I get an unexpected error:

Method "traverse" overrides class "AbstractType" in an incompatible manner
  Return type mismatch: base method returns type "CoroutineType[Any, Any, AsyncIterator[str]]", override returns type "AsyncIterator[str]"
    "AsyncIterator[str]" is not assignable to "CoroutineType[Any, Any, AsyncIterator[str]]" Pylance [reportIncompatibleMethodOverride](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportIncompatibleMethodOverride)

The following code demonstrates the error:
```python
class AbstractType(ABC):
    @abstractmethod
    async def traverse(self, url: str) -> AsyncIterator[str]:
        pass

class Implementation(AbstractType):
    async def traverse(self, url: str) -> AsyncIterator[str]:
        # Simulate some async operation
        print(f"Traversing {url}...")
        await asyncio.sleep(1)
        yield "some value"
```

This is displayed by Pylance 2025.3.2 in VS Code

**Comment by erictraut** [comment]
Yes, this behavior is correct. Refer to [this issue](https://github.com/microsoft/pyright/issues/9949) for details.

**Comment by stevenh** [comment]
Thanks for the quick reply @erictraut, my searches failed dismally to find that solution, but the link has all the info to fix, most appreciated!
